enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,487 @@
+- As an OpenShift cluster administrator, I would like to use a spread placement group to ensure that my critical
+  workloads are unlikely to suffer from hardware failures simultaneously.
+- As an operator of High-Performance Computing workloads on OpenShift, I would like to use a cluster placement group to
+  ensure that there is
+  minimal latency between instances within my cluster, allowing maximal throughput of my workloads.
+- As an operator of a large scale distributed and replicated data store, I would like to use a partition placement
+  group to ensure that different shards of my data are replicated within different fault domains within the same
+  availability zone.
+- As an OpenShift cluster administrator, I would like unneeded placement groups to be cleaned up so that I am not
+  polluting my environment with resources that are no longer in use.
+
+### API Extensions
+
+The new fields for this enhancement will be limited to the `Placement` struct within the `AWSMachineProviderConfig` in
+the Machine v1beta1 API group.
+
+The changes are outlined below:
+
+```golang
+type Placement struct {
+  // Existing Fields
+  ...
+
+  // GroupName specifies the name of an AWS placement group to create the Machine within.
+  // If the group specified does not exist, it will be created based on the GroupType
+  // paramenter.
+  // When a Machine is deleted, if this leaves an empty Placement Group, the group is also
+  // deleted.
+  // +optional
+  GroupName string `json:"groupName,omitempty"`
+
+  // GroupType specifies the type of AWS placement group to use for this Machine.
+  // This parameter is only used when a Machine is being created and the named
+  // placement group does not exist.
+  // Valid values are "Cluster", "Partition", "Spread", and omitted, which means
+  // no opinion and the platform chooses a good default which may change over time.
+  // The current default value is "Spread".
+  // +kubebuilder:validation:Enum:="Cluster";"Partition";"Spread"
+  // +optional
+  GroupType AWSPlacementGroupType `json:"groupType,omitempty"`
+
+  // PartitionCount specifies the number of partitions for a Partition placement
+  // group. This value is only observed when creating a placement group and
+  // only when the `groupType` is set to `Partition`.
+  // Note the partition count of a placement group cannot be changed after creation.
+  // If unset, AWS will provide a default partition count.
+  // This default is currently 2.
+  // +kubebuilder:validation:Minimum:=1
+  // +kubebuilder:validation:Maximum:=7
+  // +optional
+  PartitionCount int32 `json:"partitionCount,omitempty"`
+
+  // PartitionNumber specifies the numbered partition in which instances should be launched.
+  // This value is only used in conjunction with a `Partition` placement Group.
+  // It is recommended to only use this value if multiple MachineSets share
+  // a single Placement Group, in which case, each MachineSet should represent an individual partition number.
```

Comment:
cam you provide an example of using this later in the enhancement?  i think it's too big for godoc here, but I don't quite see it.

Reply:
Yep, have added a yaml example to the Partition Placement Groups heading